nasa/harmony-opendap-subsetter
# DAS-2232 - Functionality added to support SMAP L3 products

## Description

SMAP L3 does not have 1D dimension scales and grid mapping variable which is needed by HOSS to do spatial subsetting.
Methods added to override the missing grid mapping with overrides in the hoss_config.json and supporting methods.
Methods also added to use the 2D coordinate datasets to generate 1D dimension scales that could be used for to calculate
the index ranges to provide the spatial subsetted outputs 

## Jira Issue ID
DAS-2232

## Local Test Steps
- Spatial subsetting can be tested without mask fill. (till DAS-2215 is complete)
- Run Harmony in the box with HOSS. 
- Comment out of mask_fill in services.yaml file
- Run a spatial subset request locally and ensure you get the right subsetted output
- e.g. 
1) http://localhost:3000/C1268452378-EEDTEST/ogc-api-coverages/1.0.0/collections/parameter_vars/coverage/rangeset?forceAsync=true&granuleId=G1268452388-EEDTEST&subset=lat(70%3A80)&subset=lon(-160%3A-150)&variable=Soil_Moisture_Retrieval_Data_AM%2Fstatic_water_body_fraction&skipPreview=true
2) http://localhost:3000/C1268452378-EEDTEST/ogc-api-coverages/1.0.0/collections/parameter_vars/coverage/rangeset?forceAsync=true&granuleId=G1268452388-EEDTEST&subset=lat(54%3A72)&subset=lon(2%3A42)&format=application%2Fx-netcdf4&variable=Soil_Moisture_Retrieval_Data_AM%2Falbedo%2CSoil_Moisture_Retrieval_Data_AM%2Fsurface_flag&skipPreview=true

- 3D variables do not work - pending on DAS-2238
- New unit tests have not been added. The current unit tests are passing
- DAS-2236 written to handle fill values in the corners. 
- Jupyter test notebooks exist for SMAP L3 and need to be updated
- 
## PR Acceptance Checklist
* [X ] `CHANGELOG.md` updated to include high level summary of PR changes.
